From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 8 Jul 2022 16:59:49 +0200
Subject: [PATCH] HACK: don't complain about unnecessary host-tools

The cross build will fail if those tools are not available, but they are
not actually needed at buildtime.

This avoids building unnecessary stuff in host-qt6.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 qtdeclarative/tools/CMakeLists.txt | 7 +++++--
 qtquick3d/tools/CMakeLists.txt     | 2 ++
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/qtdeclarative/tools/CMakeLists.txt b/qtdeclarative/tools/CMakeLists.txt
index b86d3251a3ad..2dbbdc1e0caf 100644
--- a/qtdeclarative/tools/CMakeLists.txt
+++ b/qtdeclarative/tools/CMakeLists.txt
@@ -38,9 +38,12 @@ if(QT_FEATURE_qml_preview AND QT_FEATURE_thread AND NOT ANDROID AND NOT WASM AND
 endif()
 if(QT_BUILD_SHARED_LIBS AND QT_FEATURE_thread AND TARGET Qt::Quick AND NOT ANDROID AND NOT WASM AND NOT IOS AND NOT rtems)
     add_subdirectory(qmlscene)
-    add_subdirectory(qmltime)
+    if (NOT CMAKE_CROSSCOMPILING)
+        add_subdirectory(qmltime)
+    endif()
 endif()
 if(QT_BUILD_SHARED_LIBS
+        AND NOT CMAKE_CROSSCOMPILING
         AND QT_FEATURE_process
         AND QT_FEATURE_regularexpression
         AND QT_FEATURE_thread
@@ -68,7 +71,7 @@ if(TARGET Qt::Quick
         AND NOT rtems)
     add_subdirectory(qmleasing)
 endif()
-if(QT_FEATURE_thread AND TARGET Qt::QuickTest AND NOT ANDROID AND NOT WASM AND NOT rtems)
+if(QT_FEATURE_thread AND TARGET Qt::QuickTest AND NOT ANDROID AND NOT WASM AND NOT rtems AND NOT CMAKE_CROSSCOMPILING)
     add_subdirectory(qmltestrunner)
 endif()
 if(QT_FEATURE_private_tests AND QT_FEATURE_thread AND NOT ANDROID AND NOT WASM AND NOT IOS AND NOT rtems)
diff --git a/qtquick3d/tools/CMakeLists.txt b/qtquick3d/tools/CMakeLists.txt
index 144af64c1095..6a40010a4330 100644
--- a/qtquick3d/tools/CMakeLists.txt
+++ b/qtquick3d/tools/CMakeLists.txt
@@ -1,6 +1,7 @@
 # Copyright (C) 2022 The Qt Company Ltd.
 # SPDX-License-Identifier: BSD-3-Clause
 
+if (NOT CMAKE_CROSSCOMPILING)
 add_subdirectory(balsam)
 if(TARGET Qt::Widgets)
     if(TARGET Qt::Concurrent)
@@ -18,3 +19,4 @@ endif()
 add_subdirectory(instancer)
 add_subdirectory(shapegen)
 add_subdirectory(materialeditor)
+endif()
